tests/unit/components/Messages/Messages.test.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { render, waitFor } from "@testing-library/react-native";
+import { NavigationContainer } from "@react-navigation/native";
+
+import factory from "../../../factory";
+import Messages from "../../../../src/components/Messages/Messages";
+import { testElement } from "domutils";
+
+const mockedNavigate = jest.fn( );
+const mockMessage = factory( "RemoteMessage" );
+
+jest.mock( "../../../../src/components/Messages/hooks/useMessages" , ( ) => ( {
+  __esModule: true,
+  default: ( ) => {
+    return {
+      messageList: [mockMessage],
+      loading: false
+    };
+  }
+} ) );
+
+jest.mock( "@react-navigation/native", ( ) => {
+  const actualNav = jest.requireActual( "@react-navigation/native" );
+  return {
+    ...actualNav,
+    useNavigation: ( ) => ( {
+      navigate: mockedNavigate
+    } )
+  };
+} );
+
+jest.useFakeTimers();
+
+const renderMessages = async ( ) => waitFor(
+  ( ) => render(
+  <NavigationContainer>
+    <Messages />
+  </NavigationContainer>
+  )
+);
+
+test.todo( "add displays message subject test" );
+//unclear why this fails
+//
+// it( "displays message subject", async ( ) => {
```

Comment:
@albullington need some help here on why this doesn't work if you have a moment.

Reply:
Two things:

1) like I said, you need to change the returned value of `useMessages` to use the key messages, not messageList. If you have the wrong name of the return value, the test will fail.
2) you don't need any of the async/awaits here, since you're just rendering a component. so renderMessages should look like:

```
const renderMessages = ( ) => render(
  <NavigationContainer>
    <Messages />
  </NavigationContainer>
);
```

and similarly in the test, there should be no async/await:

```
it( "displays message subject", ( ) => {
  const { getByText } = renderMessages( );
```

if you do these two things, the test should pass.